File: solutions/python3/0053.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans = -2 ** 31
        sum = 0

        for i in range(len(nums)):
            sum += nums[i]
            if sum > ans:
                ans = sum
            if sum < 0:
                sum = 0

        return ans

File: solutions/cpp/0053.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ret = INT_MIN;
        int sum = 0;

        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            if (sum > ret) ret = sum;
            if (sum < 0) sum = 0;
        }

        return ret;
    }
};